src/core/meshcop/tlvs.hpp

```diff
@@ -1277,20 +1244,58 @@ class ChannelMask0Tlv: public ChannelMaskTlv, public ChannelMaskEntry
     }
 
     /**
+     * This method clears the bit corresponding to @p aChannel in ChannelMask.
+     *
+     * @param[in]  aChannel  The channel in ChannelMask to clear.
+     *
+     */
+    void ClearChannel(uint8_t aChannel) {
+        uint32_t aMask = HostSwap32(mMask);
+        uint8_t *mask = reinterpret_cast<uint8_t *>(&aMask);
+        mask[3 - (aChannel / 8)] &= ~(1 << (7 - (aChannel % 8)));
+        mMask = HostSwap32(*reinterpret_cast<uint32_t *>(mask));
+    }
+
+    /**
+     * This method sets the bit corresponding to @p aChannel in ChannelMask.
+     *
+     * @param[in]  aChannel  The channel in ChannelMask to set.
+     *
+     */
+    void SetChannel(uint8_t aChannel) {
+        uint32_t aMask = HostSwap32(mMask);
+        uint8_t *mask = reinterpret_cast<uint8_t *>(&aMask);
+        mask[3 - (aChannel / 8)] |= 1 << (7 - (aChannel % 8));
+        mMask = HostSwap32(*reinterpret_cast<uint32_t *>(mask));
+    }
+
+    /**
+     * This method indicates whether or not the bit corresponding to @p aChannel in ChannelMask is set.
+     *
+     * @param[in]  aChannel  The channel in ChannelMask to get.
+     *
+     */
+    bool IsChannelSet(uint8_t aChannel) const {
+        uint32_t aMask = HostSwap32(mMask);
+        const uint8_t *mask = reinterpret_cast<const uint8_t *>(&aMask);
+        return (aChannel < (GetMaskLength() * 8)) ? ((mask[3 - (aChannel / 8)] & (1 << (7 - (aChannel % 8)))) != 0) : false;
```

Comment:
I think the bit ordering could be fixed with a very minor modification to the code that was removed above.

```
bool IsChannelSet(uint8_t aChannel) const {		
    const uint8_t *mask = reinterpret_cast<const uint8_t *>(this) + sizeof(*this);		
    return (aChannel < (mMaskLength * 8)) ? ((mask[aChannel / 8] & (0x80 >> (aChannel % 8))) != 0) : false;		
}
````